### ISSUE TITLE
New noson 4.0 does not change the song

### ISSUE THREAD

**Issue by sixstone-qq** [janbar/noson-app#99]
After an upgrade to noson 4.0, once a song is ended, the same song plays again. I tried with My Index and Spotify services with the same result.

We can't shuffle anymore (the button is disabled) as well (I think that's related)

```
$ apt show noson-app
Package: noson-app
Version: 4.0~201911251226~ubuntu18.04.1
APT-Sources: http://ppa.launchpad.net/jlbarriere68/noson-app/ubuntu bionic/main amd64 Packages
```

Any help will be appreciated :)

**Comment by janbar** [comment]
Could you send me a log:

noson-app --debug 2>&1 | tee noson.log

It should show the "CurrentPlayMode" as follows:
```
(SONOS)Process: accepting new connection
(SONOS)ParseQuery: NOTIFY / HTTP/1.1
(SONOS)ParseQuery: HOST: 192.168.2.124:1400
(SONOS)ParseQuery: CONNECTION: close
(SONOS)ParseQuery: CONTENT-TYPE: text/xml
(SONOS)ParseQuery: CONTENT-LENGTH: 4850
(SONOS)ParseQuery: NT: upnp:event
(SONOS)ParseQuery: NTS: upnp:propchange
(SONOS)ParseQuery: SID: uuid:RINCON_000E58266CBC01400_sub0000000246
(SONOS)ParseQuery: SEQ: 1
(SONOS)ParseQuery: 
(SONOS)Process: TransportState = PAUSED_PLAYBACK
(SONOS)Process: CurrentPlayMode = SHUFFLE_NOREPEAT
(SONOS)Process: CurrentCrossfadeMode = 0
(SONOS)Process: NumberOfTracks = 11
(SONOS)Process: CurrentTrack = 1
(SONOS)Process: CurrentSection = 0
(SONOS)Process: CurrentTrackURI = x-sonos-http:tr%3a730593142.mp3?sid=2&flags=8224&sn=5
(SONOS)Process: CurrentTrackDuration = 0:03:19
```

**Comment by janbar** [comment]
The root cause is the playmode has been set to REPEAT_ONE with the official sonos app. I missed to handle this state.

**Comment by janbar** [comment]
Fixed in release 4.0.1

**Comment by sixstone-qq** [comment]
Awesome fix! Here is the proof of the log:
```
(SONOS)Process: TransportState = PLAYING
(SONOS)Process: CurrentPlayMode = REPEAT_ONE
(SONOS)Process: CurrentCrossfadeMode = 0
(SONOS)Process: NumberOfTracks = 30
(SONOS)Process: CurrentTrack = 4
```
The update to noson 4.1 fixed the issue.

Thanks a lot!! :)
